source/fe/fe_series_fourier.cc

```diff
@@ -179,16 +189,23 @@ namespace FESeries
   Fourier<dim, spacedim>::Fourier(
     const std::vector<unsigned int> &      n_coefficients_per_direction,
     const hp::FECollection<dim, spacedim> &fe_collection,
-    const hp::QCollection<dim> &           q_collection)
+    const hp::QCollection<dim> &           q_collection,
+    const unsigned int                     component)
     : n_coefficients_per_direction(n_coefficients_per_direction)
     , fe_collection(&fe_collection)
     , q_collection(q_collection)
     , fourier_transform_matrices(fe_collection.size())
+    , component(component)
   {
     Assert(n_coefficients_per_direction.size() == fe_collection.size() &&
              n_coefficients_per_direction.size() == q_collection.size(),
            ExcMessage("All parameters are supposed to have the same size."));
 
+#ifdef DEBUG
+    for (unsigned int i = 0; i < fe_collection.size(); ++i)
+      AssertIndexRange(component, fe_collection[i].n_components());
+#endif
```

Comment:
Well placed assertion! :-)

Reply:
But I think you don't actually need it in this generality. `hp::FECollection` enforces that all elements have the same number of vector components, so it is probably enough to just check for `[0]`.